myokit/formats/diffsl/_exporter.py

```diff
@@ -0,0 +1,523 @@
+
+        pace = model.binding('pace')  # Reserve 'Vc' for pace variable
+        if pace is not None:
+            needs_renaming['Vc'] = []
+
+        # Find naming conflicts, create inverse mapping
+        name_to_var = collections.OrderedDict()
+        for var, name in var_to_name.items():
+
+            # Known conflict?
+            if name in needs_renaming.keys():
+                needs_renaming[name].append(var)
+                continue
+
+            # Test for new conflicts
+            var2 = name_to_var.get(name, None)
+            if var2 is not None:
+                needs_renaming[name] = [var2, var]
+                continue
+
+            name_to_var[name] = var
+
+        # Resolve naming conflicts
+        for name, variables in needs_renaming.items():
+            # Add a number to the end of the name, increasing it until it's
+            # unique
+            i = 1
+            root = name
+            for var in variables:
+                name = f'{root}{uscore}{i}'
+                while name in name_to_var:
+                    i += 1
+                    name = f'{root}{uscore}{i}'
+                var_to_name[var] = name
+                name_to_var[name] = var
+
+        if pace is not None:
+            var_to_name[pace] = 'Vc'
+
+        return var_to_name
+
+    def _get_time_factor(self, time_var):
+        """
+        Get factor to convert time to ms.
+        time_units * time_factor means going from time_units to ms.
+        """
+        time_unit = time_var.unit()
+
+        if time_unit is None:
+            return None, None
+
+        if time_unit == myokit.units.ms:
+            return None, None
+
+        time_factor = time_factor_inv = None
+
+        try:
+            time_factor = myokit.Number(
+                myokit.Unit.conversion_factor(time_unit, myokit.units.ms)
+            )
+            time_factor_inv = myokit.Number(
+                1 / time_factor.eval(), 1 / time_factor.unit()
+            )
+        except myokit.IncompatibleUnitError:
+            warnings.warn(
+                'Unable to convert time units '
+                + str(time_unit)
+                + ' to recommended units of ms.'
+            )
+
+        return time_factor, time_factor_inv
+
+    def _guess_currents(self, model):
+        """
+        Tries to make a list of membrane currents. Removes potentials
+        from the guessed list.
+        """
+        unmatch_units = [
+            myokit.units.mV,
+            myokit.units.V,
+        ]
+        guessed_currents = guess.membrane_currents(model)
+        return [x for x in guessed_currents if x.unit() not in unmatch_units]
+
+    def _guess_membrane_potential(self, model):
+        """
+        Tries to find the membrane potential. If it's not a state, converts it
+        to one so that expressions depending on it are not seen as constants.
+        """
+        vm = guess.membrane_potential(model)
+        if not vm.is_state():
```

Comment:
Why is this bit necessary? Seems a bit drastic, especially as the Vm is a guess?

Reply:
Removed in 1a212493.